app/workers/metrics/record_db_and_es_record_counts_worker.rb

```diff
@@ -0,0 +1,64 @@
+# This worker checks the number of records for each Elasticsearch index and
+# compares it to the number of records in our database.
+#
+# For indexes that don't match a single model (no model or multiple models) we
+# need to implement a custom db_count method on the Search class to do the
+# counts for us
+module Metrics
+  class RecordDbAndEsRecordCountsWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :low_priority, retry: 5
+
+    # Once the Elasticsearch migration is complete we can update this value to
+    # Search::Cluster::SEARCH_CLASSES which will include all indexes.
+    SEARCH_CLASSES = [
+      Search::ChatChannelMembership,
+      Search::ClassifiedListing,
+      Search::Tag,
+    ].freeze
+
+    # Adjustable margin of error - this is how far off the index count can be
+    # from the database count before we raise an error
+    def perform(margin_of_error = 0)
+      SEARCH_CLASSES.each do |search_class|
+        margin_of_error = margin_of_error.to_i
+        db_count = db_count(search_class)
+        index_count = Search::Client.count(index: search_class::INDEX_ALIAS).dig("count")
+        record_difference = (db_count - index_count).abs
+
+        tags = {
+          search_class: search_class,
+          db_count: db_count,
+          index_count: index_count,
+          record_difference: record_difference,
+          margin_of_error: margin_of_error,
+          action: "record_count"
+        }
```

Comment:
I figured it can't hurt to send all the data as tags so we can debug very quickly and make some granular monitors in Datadog.